### ISSUE TITLE
Develop common version catalogs

### ISSUE THREAD

**Issue by ak3216** [airshipit/airshipctl#316]
**Problem description (if applicable)**
As an operator, I would like to have a mechanism for specifying software/component versions in a common place so I can manage the platform in a consist fashion. Additionally, I would like to have the ability to override a version value at a lower type/site level.

**Proposed change**
1. Identify all function instances where software/component versions are being specified. This should not include the k8s apiVersion.
2. Based on the composite/phase approach, determine at what level the version catalog should be specified.  
3. Create replacement rules that update the version values into the respective functions.
Note: CAPI component versions themselves may be an outlier – they’re encoded in the directory name structure (although also reflected in many container image tags), and are closely tied to the version of airshipctl itself.  There may need to be an alternate mechanism to override capi component versions, or we might decide not to override capi versions.  
4. Once this is in place for airshipctl, the next steps will be to create issues to make one or more similar version catalogues/replacement rules in each of treasuremap & opentack-helm-tbd-repo.

**Comment by mattmceuen** [comment]
I've started documenting our versions catalogue approach (as well as the rest of laying strategy, since folks have been asking) [here](https://review.opendev.org/#/c/745974/1).  The unique tasks for bootstrapping versions substitution are:

1. Create a `versions-airshipctl.yaml` in a new function `catalogues-airshipctl` in the airshipctl project
2. Extrapolate all container versions from airshipctl function into this resource, taking inspiration from Airship1.
3. Extrapolate all relevant disk image references into the catalogue.  Should just be the xenial-server-cloudimg-amd64-disk1.img
in the k8scontrol function (the uefi_esp.img in the `baremetal-operator` function appears to be n/a)
4. There are no chart references in airshipctl to extract, but let's put a placeholder in there for dex
5. Extrapolate the kubernetes version (v1.17.0) into `versions-airshipctl`. Note: let's leave as-is the k8s versions embedded in the apt-get commands within a couple functions, since those will go away with the image builder.
6. Add deferred replacement rules to the `baremetal-operator` function (for container images), the `k8scontrol` function (for the disk image and the kubernetes version in the controlplane spec).

Clusterctl-managed resources require special treatment:
7. Move the `shared/clusterctl` entrypoint out of `test-site` into its a standalone clusterctl function.
8. Add deferred replacement rules to the `clusterctl` function that source containers from `versions-airshipctl`, and jam them into
variables in the `Clusterctl` resource.  Clusterctl supports a couple different forms of variables, so review those first.
9. Edit the CAPI component functions' contents as necessary to consume the variables from `Clusterctl`

10. Update the `test-site` definition to pull in & apply deferred replacement rules defined above
11. Update the `airship-core` type definition in treasuremap to pull in & apply the deferred replacement rules defined above

**Comment by mattmceuen** [comment]
A changeset addressing 1-6 above: https://review.opendev.org/#/c/746608

**Comment by mattmceuen** [comment]
This one addresses 7-10: https://review.opendev.org/#/c/746764/

**Comment by mattmceuen** [comment]
This one wraps up #11 (Treasuremap): https://review.opendev.org/#/c/752101/
